### PR TITLE
Add support to M5 ATOMIC PortABC Extension Base

### PR DESCRIPTION
**Pull request by oxbown**
Two new build versions for Atom and AtomS3 with port ABC 
Commented #58 
Also encreased i2c tasksize #61

### CODE REVIEW

**Comment by wellenvogel** [comment]
Thanks for building and testing this.
Anyway for the support of new hardware I would like to avoid the definitions of new environments for each combination but instead going with the approach of defining building blocks and adding them to the build config (build.yaml) so that you can select the necessary pieces in the build UI.
So the support for the port ABC would need to become a "KIT" definition - like https://github.com/wellenvogel/esp32-nmea2000/blob/e1044821a7b61de3e1c393986262f44952bffddd/lib/hardware/GwHardware.h#L159.
And we would need some config options for the build UI.
Basically I would assume to have serial/can/i2c as potential grove users.
Most probably it should be iic2 and serial2.

**Comment by oxbown** [comment]
Sure, I´ll test it!
Feedback will come.

**Comment by oxbown** [comment]
Testing the actual revision i found the same ussie as here https://github.com/wellenvogel/esp32-nmea2000/issues/61
Stack canary triggered. The setup is: AtomS3+CAN+GPS+RS485+ENV3
`17:36:41.699 -> E (2370) esp_core_dump_flash: Core dump flash config is corrupted! CRC=0x7bd5c66f instead of 0x0
17:36:41.732 -> Rebooting...
17:36:41.732 -> ESP-ROM:esp32s3-20210327
17:36:41.732 -> Build:Mar 27 2021
17:36:41.732 -> rst:0xc (RTC_SW_CPU_RST),boot:0x2b (SPI_FAST_FLASH_BOOT)
17:36:41.732 -> Saved PC:0x40377870
17:36:41.732 -> SPIWP:0xee
17:36:41.732 -> mode:DIO, clock div:1
17:36:41.732 -> load:0x3fce3808,len:0x44c
17:36:41.732 -> load:0x403c9700,len:0xbe4
17:36:41.732 -> load:0x403cc700,len:0x2a38
17:36:41.732 -> entry 0x403c98d4
17:36:41.937 -> E (231) esp_core_dump_flash: No core dump partition found!
17:36:41.937 -> E (231) esp_core_dump_flash: No core dump partition found!
17:36:41.970 -> [   246][E][Preferences.cpp:50] begin(): nvs_open failed: NOT_FOUND
17:36:42.220 -> LOG:400:GWWIFI: AP auto shutdown disabled (0s)
17:36:42.220 -> GWSERIAL:505:initialized IIC 1 at sda=5,scl=6
17:36:42.220 -> GWSERIAL:506:wifi AP pass: esp32nmea2k
17:36:42.220 -> GWSERIAL:506:admin pass: esp32admin
17:36:44.000 -> Guru Meditation Error: Core  0 panic'ed (Unhandled debug exception). 
17:36:44.000 -> Debug exception reason: Stack canary watchpoint triggered (iicTask) 
17:36:44.000 -> Core  0 register dump:
17:36:44.000 -> PC      : 0x40380ce7  PS      : 0x00060b36  A0      : 0x80383944  A1      : 0x3fcc4fe0  
17:36:44.000 -> A2      : 0x3fce9f14  A3      : 0xb33fffff  A4      : 0x0000abab  A5      : 0x00060b23  
17:36:44.000 -> A6      : 0x00060b20  A7      : 0x0000cdcd  A8      : 0xb33fffff  A9      : 0xffffffff  
17:36:44.035 -> A10     : 0x3fcc53e0  A11     : 0x00000000  A12     : 0x7ae147ae  A13     : 0x404cee14  
17:36:44.035 -> A14     : 0x02ce9f14  A15     : 0x00ffffff  SAR     : 0x0000000a  EXCCAUSE: 0x00000001  
17:36:44.035 -> EXCVADDR: 0x00000000  LBEG    : 0x400556d5  LEND    : 0x400556e5  LCOUNT  : 0xffffffff  
17:36:44.035 -> 
17:36:44.035 -> 
17:36:44.035 -> Backtrace: 0x40380ce4:0x3fcc4fe0 0x40383941:0x3fcc5020 0x40383bad:0x3fcc5040 0x403780a9:0x3fcc5060 0x403780bd:0x3fcc5090 0x403780ea:0x3fcc50b0 0x40383f91:0x3fcc50d0 0x420bb2f6:0x3fcc50f0 0x420b53f2:0x3fcc5190 0x420b2ad5:0x3fcc54a0 0x42032eee:0x3fcc5560 0x4202b22f:0x3fcc55d0 0x4202874e:0x3fcc5660 0x42006bc6:0x3fcc56b0 0x42006db6:0x3fcc5710 0x4203037e:0x3fcc5730 0x420aa4db:0x3fcc5760 0x420aa73d:0x3fcc58c0 0x420c7765:0x3fcc5910 0x420a8f65:0x3fcc5930 0x420c568d:0x3fcc59d0
`

This is the log of the online build:
`GIT_SHA="bfc6af1f89bf7a0a931e9b82734fe2f201d50a76" PLATFORMIO_BUILD_FLAGS=" -DM5_PORTABC -DSERIAL_GROOVE_485_A -DM5_GPS_UNIT_B -DM5_ENV3_C -DM5_CANUNIT -DGWRELEASEVERSION=testci20240321" pio run -e m5stack-atoms3-generic 
`

**Comment by wellenvogel** [comment]
Really interesting.
Was running nearly the same set up on an atom lite without issues.
So I have to increase the stack size...

**Comment by wellenvogel** [comment]
Maybe you could give it a second try with testci20240322 - iic task stack size now at 4000.

**Comment by oxbown** [comment]
@wellenvogel online build service not show "PortABC" option

**Comment by oxbown** [comment]
> Maybe you could give it a second try with testci20240322 - iic task stack size now at 4000.

Actual testci20240322  tag not compile in circle ci, maybe the last commit break something.

**Comment by wellenvogel** [comment]
ahhh, should have been testing also with S3...

**Comment by wellenvogel** [comment]
ok new tag: testci20240323

**Comment by oxbown** [comment]
Build OK! Seems to work well on the esp32s3

**Comment by oxbown** [comment]
Many thanks for your awesome work!
